Source/help.cpp

```diff
@@ -138,6 +139,29 @@ int NumVisibleLines()
 	return (ContentsTextHeight() - 1) / LineHeight() + 1; // Ceil
 }
 
+void DrawHelpSlider(const Surface &out)
+{
+	const Point uiPosition = GetUIRectangle().position;
+	const int sliderXPos = ContentTextWidth + uiPosition.x + 36;
+	const Point sliderPosition = uiPosition;
+	int sliderStart = uiPosition.y + HeaderHeight() + LineHeight() + 3;
+	int sliderEnd = uiPosition.y + PaddingTop + PanelHeight - 12;
+	ClxDraw(out, { sliderXPos, sliderStart }, (*pSTextSlidCels)[11]);
+	sliderStart += 12;
+	int sliderCurrent = sliderStart;
+	for (; sliderCurrent < sliderEnd; sliderCurrent += 12) {
+		ClxDraw(out, { sliderXPos, sliderCurrent }, (*pSTextSlidCels)[13]);
+	}
+	ClxDraw(out, { sliderXPos, sliderCurrent }, (*pSTextSlidCels)[10]);
+	// Subtract visible lines from the total number of lines to get the actual
+	// scroll range
+	const float scrollRange = HelpTextLines.size() - NumVisibleLines();
+	// Subtract the size of the arrow buttons to get the length of the interior
+	// part of the slider
+	const float sliderLength = sliderCurrent - 12 - sliderStart;
+	ClxDraw(out, { sliderXPos, sliderStart + (int)(((float)SkipLines * sliderLength) / scrollRange) }, (*pSTextSlidCels)[12]);

```

Comment:
I'm not sure I see the point in having a float here, it's just subtracting and multiplying whole numberes. The cast back to int happens right after the division so the precision is discarded right away.

Reply:
You're 100% right, I think I wanted to do it some other way at first and thought for some reason we needed the cast, but we really don't.

Reply:
Ok, if you change that to integer math then I'm happy to merge this :)

Reply:
Should I add a commit to the PR ? Or is modifying the commit cleaner ? (I'm new to this as mentioned in the issue ahah)

Reply:
We prefer having things clean, but that also means that we will squash your PR in to one commit if it's messy so either is fine :)

Reply:
```suggestion
	const int sliderLength = sliderCurrent - 12 - sliderStart;
	ClxDraw(out, { sliderXPos, sliderStart + (SkipLines * sliderLength / scrollRange) }, (*pSTextSlidCels)[12]);
```

Reply:
Done :). Thank you for your quick answers !